### DISCUSSION TITLE
How to work with router.isFallback on production builds?

### DISCUSSION THREAD

**Discussion by gglnx** [vercel/next.js#11240] [Help]
I'm using next.js 9.3 and `getStaticPaths` with `fallback: true`. In my page component I'm checking for `isFallback` and rendering a spinner on fallback and an 404 error page if props are missing:

```js
const Location: React.FC<LocationProps> = ({ location }) => {
  const router = useRouter();

  // Fallback
  if (router.isFallback) {
    return (
      <Spinner />
    );
  }

  // No location found
  if (!location) {
    return <Error statusCode={404} />;
  }
```

While development opening an uncached pages shows the spinner and after loading everything my component, but on production the 404 error page is shown instead of the spinner.

How can I make sure that my component is build with the spinner as default?

**Comment by Timer** [comment]
How are you deploying your application?

**Reply by gglnx** [reply]
Git checkout on the server, then `next build` and pm2 with `next start`.
